apps/laboratory/src/components/Ethers/EthersSendCallsTest.tsx

```diff
@@ -19,12 +20,28 @@ export function EthersSendCallsTest() {
   const { walletProvider } = useWeb3ModalProvider()
   const toast = useChakraToast()
 
-  const atomicBatchSupportedChains =
-    address && walletProvider instanceof EthereumProvider
-      ? getCapabilitySupportedChainInfo(WALLET_CAPABILITIES.ATOMIC_BATCH, walletProvider, address)
-      : []
+  const [atomicBatchSupportedChains, setAtomicBatchSupportedChains] = useState<
+    Awaited<ReturnType<typeof getCapabilitySupportedChainInfo>>
+  >([])
 
-  const atomicBatchSupportedChainNames = atomicBatchSupportedChains
+  const [lastCallsBatchId, setLastCallsBatchId] = useState<string | null>(null)
+
+  useEffect(() => {
+    if (
+      address &&
+      (walletProvider instanceof EthereumProvider || walletProvider instanceof W3mFrameProvider)
```

Comment:
does this checking through `instanceof` may be problematic through different versions of the `@walletconnect/ethereum-provider`? We got this issue before